crosschain/simple/src/main/java/tech/pegasys/samples/crosschain/simple/transaction/CrosschainTransactionNoParams.java

```diff
@@ -1,4 +1,331 @@
+        LOG.info(" Contract 1 deployed on sidechain 1 (id={}), at address: {}", SC1_SIDECHAIN_ID, contract1Address);
+    }
+
+
+
+
+    private void core() throws Exception {
+        LOG.info("Running Core Part of Sample Code");
+
+        LOG.info(" Set state in each contract to known values that aren't zero.");
+        TransactionReceipt transactionReceipt;
+        LOG.info("  Single-chain transaction: Contract2.clear()");
+        transactionReceipt = this.contract2.clear().send();
+        assertTrue(transactionReceipt.isStatusOK());
+
+        checkExpectedValues(false);
+
+        Scanner myInput = new Scanner( System.in );
+        Boolean c2Val = this.contract2.get().send();
+
+        LOG.info("  Executing call simulator to determine parameter values and expected results");
+        CallSimulator sim = new CallSimulator();
+        sim.c2clear();
+        sim.c1Crosschain_setter();
+
+        LOG.info("  Constructing Nested Crosschain Transaction");
+        // Originating sidechain is sidechain 1.
+        CrosschainContextGenerator contextGenerator = new CrosschainContextGenerator(SC1_SIDECHAIN_ID);
+
+        // Call to contract 2
+        // Contract 2 is called by contract 1 on sidechain 1.
+        CrosschainContext subordinateContext = contextGenerator.createCrosschainContext(SC1_SIDECHAIN_ID, this.contract1Address);
+        byte[] subordinateTransactionC2 = this.contract2.set_AsSignedCrosschainSubordinateTransaction(subordinateContext);
+
+        // Call to contract 1
+        byte[][] subordinateTransactionsAndViewsForC1;
+        subordinateTransactionsAndViewsForC1 = new byte[][] {subordinateTransactionC2};
+
+        LOG.info("  Executing Crosschain Transaction");
+        // The originating transaction starts in Contract 1.
+        subordinateContext = contextGenerator.createCrosschainContext(subordinateTransactionsAndViewsForC1);
+        transactionReceipt = this.contract1.crosschain_setter_AsCrosschainTransaction(subordinateContext).send();
+        LOG.info("  Transaction Receipt: {}", transactionReceipt.toString());
+        assertTrue(transactionReceipt.isStatusOK());
+
+        boolean stillLocked;
+        final int tooLong = 10;
+        int longTimeCount = 0;
+        StringBuffer graphicalCount = new StringBuffer();
+        do {
+            longTimeCount++;
+            if (longTimeCount > tooLong) {
+                LOG.error("Contract {} did not unlock", this.contract1Address);
+            }
+            Thread.sleep(500);
+            CrosschainIsLocked isLockedObj = this.web3jSc1.crosschainIsLocked(this.contract1Address, DefaultBlockParameter.valueOf("latest")).send();
+            stillLocked = isLockedObj.isLocked();
+            if (stillLocked) {
+                graphicalCount.append(".");
+                LOG.info("   Waiting for the contract to unlock{}", graphicalCount.toString());
+            }
+        } while (stillLocked);
+
+        checkExpectedValues(sim.c2Get());
```

Comment:
though I guess having a simulator makes this simpler...

Reply:
Leaving it in as a vestigial simulator, as discussed.